Steps 1-4: About Me page

jhuang09 opened the pull request:
- Added a servlet for the "About Me" page 
- Created an about.html page

jhuang09 committed:
About Me step 2: Adding servlet and creating about.html

dhgarrette reviewed (changes requested):
Thanks, Janny!  Looks pretty good!  Let's adjust the formatting slightly to match the existing code: java and html should use 2-space indentation.  (I realize the code in the instructions has it wrong, but that's ok, we can fix it here.)  You can also try the formatter tool available here: https://github.com/google/google-java-format

jhuang09 committed:
About Me step 3: added more interactive servlet

jhuang09 committed:
updated spacing in AboutMeServlet.java and about.html files

jhuang09 commented:
I've updated the spacing for the two files! Let me know what you think. :)

dhgarrette commented on src/main/java/com/google/codeu/servlets/AboutMeServlet.java:
```diff
@@ -0,0 +1,65 @@
+  
+    String user = request.getParameter("user");
+  
+    if(user == null || user.equals("")) {
```
One space between `if` and `(`.

dhgarrette replied:
`user.equals("")` can be written more simply as `user.isEmpty()`.

dhgarrette commented on src/main/java/com/google/codeu/servlets/AboutMeServlet.java:
```diff
@@ -0,0 +1,65 @@
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
```
Four spaces `throws` since it's a continuation of the previous line.

dhgarrette commented on src/main/java/com/google/codeu/servlets/AboutMeServlet.java:
```diff
@@ -0,0 +1,65 @@
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
```
Same here: 4 spaces.

dhgarrette reviewed (changes requested):
Cool.  A couple more comments now.

jhuang09 committed:
updated spacing in AboutMeServlet.java

jhuang09 commented:
I've made some changes according to your comments! I also didn't realize that I added the other files I've made changes to, but I added some parts to the html and js file.

dhgarrette commented on src/main/webapp/js/user-page-loader.js:
```diff
@@ -67,6 +67,22 @@ function fetchMessages() {
+    }
+    
+    aboutMeContainer.innerHTML = aboutMe;
+
```
You can remove this blank line.

dhgarrette commented on src/main/webapp/js/user-page-loader.js:
```diff
@@ -67,6 +67,22 @@ function fetchMessages() {
+
+  });
+}
+
```
And this one.

dhgarrette commented on src/main/webapp/js/user-page-loader.js:
```diff
@@ -67,6 +67,22 @@ function fetchMessages() {
       });
 }
 
+function fetchAboutMe(){
```
It looks like this isn't being called anywhere yet.

dhgarrette commented on src/main/webapp/about.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Is the plan to have "about me" content show up on both `about.html` and `user-page.html`?

jhuang09 replied:
Now that I'm rereading the specs, the "about me" content should show up on just user-page.html. I will delete the about.html!

dhgarrette commented:
No problem.  Yeah, if you commit and push to a branch that has an open PR, then those changes will become part of that PR.  If you want to make changes without affecting the PR, then you'll do those in a branch of your branch (which can then become its own PR later).

jhuang09 committed:
deleted about.html, updated js and html where a user's About Me form is visible only to themselves.

jhuang09 commented:
Hi again! I've removed about.html and have made the changes you've suggested. I have also changed some bits in the js file so that the function fetchAboutMe() will get called and that the "About Me" form will only be visible to the user when they are logged in. When you have the chance, let me know what you think!

dhgarrette commented on src/main/webapp/js/user-page-loader.js:
```diff
@@ -67,6 +68,22 @@ function fetchMessages() {
+      })
+      .then((aboutMe) => {
+        const aboutMeContainer = document.getElementById('about-me-container');
+        if(aboutMe == ''){
```
Formatting: ` if (aboutMe == '') {`

dhgarrette commented on src/main/webapp/js/user-page-loader.js:
```diff
@@ -67,6 +68,22 @@ function fetchMessages() {
 }
 
+/** Fetches about me and displays onto page. */
+function fetchAboutMe(){
```
Formatting: `fetchAboutMe() {`

dhgarrette reviewed (approved):
A couple more tiny formatting things, but otherwise it looks great!  A very nice first PR, Janny!

jhuang09 committed:
fixed spaces! woohoo!
